src/commands/schema_diff.ts

```diff
@@ -0,0 +1,123 @@
+import { BranchScopeProps } from '../types';
+import { createPatch } from 'diff';
+import { Database } from '@neondatabase/api-client';
+import chalk from 'chalk';
+import { writer } from '../writer.js';
+import { branchIdFromProps, branchIdResolve } from '../utils/enrichers.js';
+
+type SchemaDiffProps = BranchScopeProps & {
+  branch?: string;
+  baseBranch?: string;
+  compareBranch: string;
+  database: string;
+};
+
+const COLORS = {
+  added: chalk.green,
+  removed: chalk.red,
+  header: chalk.yellow,
+  section: chalk.magenta,
+};
+
+type ColorId = keyof typeof COLORS;
+
+export const schemaDiff = async (props: SchemaDiffProps) => {
+  props.branch = props.baseBranch || props.branch;
```

Comment:
do you still need this line?

Reply:
ah, because of context?

Reply:
yes, for the context. Maybe I can move this line to middleware in next PR